### PR TITLE
osd: make operations on ReplicatedBackend::in_progress_ops more effective

### PR DESCRIPTION
**Pull request by ifed01**
… effective.

Signed-off-by: Igor Fedotov <ifedotov@suse.com>

### CODE REVIEW

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2017-11-21_16:17:37-rados-wip-kefu-testing-2017-11-21-1844-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2017-11-22_05:41:59-rados-wip-kefu-testing-2017-11-21-1844-distro-basic-mira/

File: src/osd/ReplicatedBackend.cc [hunk 1]
```diff
@@ -246,12 +246,13 @@ void ReplicatedBackend::on_change()
   dout(10) << __func__ << dendl;
   for (map<ceph_tid_t, InProgressOp>::iterator i = in_progress_ops.begin();
        i != in_progress_ops.end();
-       in_progress_ops.erase(i++)) {
+       i++) {
```
Comment (tchaikov) [inline src/osd/ReplicatedBackend.cc]: might want to use `++i`, as `i` is not a primitive type. will do this in a follow-up PR, though.

**Comment by xiexingguo** [comment]
Wonder why the `reviewed-by: ` lines are not right...
@tchaikov Do you manually add those lines or use the ptl-tool script?
Please ignore if you do it manually :-)

**Comment by tchaikov** [comment]
i do it manually, sorry for not putting your name in the "Reviewed-by" =(

File: src/osd/ReplicatedBackend.cc (hunk 1 above)
Comment (tchaikov) [reply src/osd/ReplicatedBackend.cc]: #19097

**Comment by xiexingguo** [comment]
> i did it manually, sorry for not putting your name in the "Reviewed-by" =(

Never mind :-)   I see folks are switching to use the new "ptl-tool" for auto-merging, maybe you can also have a try of that \o/

**Pull request merged**